packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -634,10 +676,20 @@ abstract class DiagnosticsNode {
   /// `:` is typically used as a separator when displaying as text.
   final bool showSeparator;
 
+  /// Whether the diagnostics should be filtered when only showing diagnostics
+  /// with at least the specified level.
+  bool isFiltered(DiagnosticLevel minLevel) {
+    return level.index < minLevel.index || hidden;
+  }
+
   /// Whether the diagnostics should be hidden when showing the default
   /// view of a tree.
```

Comment:
this should talk about its relationship with level

Reply:
I realized this method would be cleaner to document if I removed the `hidden` property from DiagnosticNode. Added [DiagnosticLevel.hidden] to serve the same role. This makes filtering more standard. I also cleaned up toString output for FlagProperty and ObjectFlagProperty cases where the output previously didn't matter as the properties were hidden.

One open question is whether I should make the breaking change to remove the hidden constructor argument as well. It is redundant but is also convenient shorthand for toggling whether a property should have DiagnosticLevel.hidden.